### PR TITLE
Authentication Service (unit tests)

### PR DESCRIPTION
**Pull request by jasonSchroeder89**
@GattieTime @KennethDavis391

Created AuthService Angular Service, wrote unit tests for unimplemented methods. Methods that require valid login credentials are not unit-tested for security reasons.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,19 +16,32 @@
...
     "zone.js": "~0.10.2"
   },
   "devDependencies": {
+<<<<<<< HEAD
     "@angular-devkit/build-angular": "~0.1100.5",
     "@angular/cli": "~11.0.5",
+=======
+    "@angular-devkit/build-angular": "^0.1100.7",
+    "@angular/cli": "^11.0.7",
+>>>>>>> auth
```
Comment (GattieTime) [inline package.json]: Do not commit unresolved merge conflicts. Also please do not make updates to the dependencies while in the sprint unless required for functionality. Updates can be their own PRs after the sprint completes.

File: src/app/app.component.html [hunk 2]
```diff
@@ -532,3 +533,8 @@ <h2>Next Steps</h2>
   
   
   <router-outlet></router-outlet>
+=======
```
Comment (GattieTime) [inline src/app/app.component.html]: merge conflict.

File: src/app/app.module.ts [hunk 3]
```diff
@@ -1,20 +1,35 @@
 import { BrowserModule } from '@angular/platform-browser';
 import { NgModule } from '@angular/core';
-
 import { AppRoutingModule } from './app-routing.module';
 import { AppComponent } from './app.component';
 import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+<<<<<<< HEAD
```
Comment (GattieTime) [inline src/app/app.module.ts]: merge conflict

File: src/app/app.module.ts [hunk 4]
```diff
@@ -1,20 +1,35 @@
...
-    AppComponent
+    AppComponent,
+    HeaderComponent,
+    LoginComponent
   ],
   imports: [
     BrowserModule,
     AppRoutingModule,
     BrowserAnimationsModule,
+<<<<<<< HEAD
```
Comment (GattieTime) [inline src/app/app.module.ts]: merge conflict

**Review by GattieTime — CHANGES_REQUESTED** [review]
Please resolve merge conflicts then I will review.

File: package.json (hunk 1 above)
Comment (jasonSchroeder89) [reply package.json]: Resolved conflict by removing changed dependency version.

File: src/app/app.component.html (hunk 2 above)
Comment (jasonSchroeder89) [reply src/app/app.component.html]: Resolved merge conflict.

File: src/app/app.module.ts (hunk 3 above)
Comment (jasonSchroeder89) [reply src/app/app.module.ts]: Resolved merge conflict.

File: src/app/app.module.ts (hunk 4 above)
Comment (jasonSchroeder89) [reply src/app/app.module.ts]: Resolved merge conflict.

**Review by GattieTime — APPROVED** [review]
